### PR TITLE
Added unique MayBuyConstructionsInPupets

### PR DESCRIPTION
**Pull request by rpolitex**
It's a tiny update that adds the Unique allowed-to-buy items (units and buildings) in puppet cities.

May be helpful for Modders to create more versatile nations.

(This is my first PR to Unciv, so any suggestions are welcomed).

### CODE REVIEW

**Comment by SeventhM** [comment]
On a separate note, I hate to be that person, but we're missing a StateForConditionals

**Comment by SeventhM** [comment]
> Civilization.hasUnique **has** a default

Wait, what? I didn't even know Civilization had a default. I don't remember cities having one. Maybe that's worthwhile. Hmm...

For city related stuff, StateForConditionals(city.civ, city) is always ideal for stuff like "in cities with a [buildinName]"

**Comment by rpolitex** [comment]
@SomeTroglodyte Thanks for responce.

> But is such a Unique really meaningful?

The main back behind this PR is to help implement nations such a [Venetians](https://civilization.fandom.com/wiki/Venetian_(Civ5)).  They cannot annex cities and cannot build new cities on their own. 
However, they can buy things in Puppet cities.

> Also, will buying a Courthouse "prematurely" process happiness correctly?

Good catch! Worse to check.

**Comment by yairm210** [comment]
Well then, that makes it a *base game functionality*! Important context information!

File: core/src/com/unciv/models/ruleset/unique/UniqueType.kt [hunk 1]
```diff
@@ -240,6 +240,7 @@ enum class UniqueType(
     TriggersCulturalVictory("Triggers a Cultural Victory upon completion", UniqueTarget.Global),
 
     /// Misc.
+    MayBuyConstructionsInPupets("May buy items in puppet cities", UniqueTarget.Global),
```
Comment (yairm210) [inline core/src/com/unciv/models/ruleset/unique/UniqueType.kt:243]: Rename 'pupet' to 'puppet'

File: core/src/com/unciv/ui/screens/cityscreen/CityConstructionsTable.kt [hunk 2]
```diff
@@ -149,7 +149,10 @@ class CityConstructionsTable(private val cityScreen: CityScreen) {
 
     private fun updateButtons(construction: IConstruction?) {
         buyButtonsTable.clear()
-        if (!cityScreen.canCityBeChanged()) return
+        if (!cityScreen.canChangeState) return
+        /** [UniqueType.MayBuyConstructionsInPupets] support - we need a buy button for civs that could buy items in puppets */
+        if (cityScreen.city.isPuppet && !cityScreen.city.civ.hasUnique(UniqueType.MayBuyConstructionsInPupets)) return
```
Comment (yairm210) [inline core/src/com/unciv/ui/screens/cityscreen/CityConstructionsTable.kt:154]:
```suggestion
        if (cityScreen.city.isPuppet && !cityScreen.city.getMatchingUniques(UniqueType.MayBuyConstructionsInPupets).any()) return 
```

That way, it contains the state for conditionals seventhm was talking about, *and* supports city-specific uniques, such as from religions

**Comment by rpolitex** [comment]
@yairm210  fixed your comments. pls review again

**Review by yairm210 — APPROVED** [review]
Lgtm

**Pull request merged**